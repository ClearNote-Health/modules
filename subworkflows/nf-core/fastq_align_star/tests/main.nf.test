nextflow_workflow {

    name "Test Subworkflow FASTQ_ALIGN_STAR"
    script "../main.nf"
    workflow "FASTQ_ALIGN_STAR"

    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "subworkflows/fastq_align_star"
    tag "star/align"
    tag "star/genomegenerate"
    tag "rsem/preparereference"
    tag "subworkflows/bam_sort_stats_samtools"

    setup {
        run("STAR_GENOMEGENERATE") {
            script "../../../../modules/nf-core/star/genomegenerate/main.nf"
            process {
                """
                input[0] = Channel.of([
                    [ id:'test_fasta' ],
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true) ]
                ])
                input[1] = Channel.of([
                    [ id:'test_fai' ],
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta.fai', checkIfExists: true) ]
                ])
                input[2] = Channel.of([
                    [ id:'test_gtf' ],
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true) ]
                ])
                """
            }
        }
    }

    test("homo_sapiens - fastq - single_end") {
        config "./nextflow.config"
        when {
            workflow {
                """
                input[0] = Channel.of([
                    [ id:'test', single_end:true ], // meta map
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_1.fastq.gz', checkIfExists: true) ]
                ])
                input[1] = STAR_GENOMEGENERATE.out.index
                input[2] = Channel.of([
                    [ id:'test_gtf' ], // meta map
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true) ]
                ])
                input[3] = true       // star_ignore_sjdbgtf
                input[4] = 'illumina' // seq_platform
                input[5] = false      // seq_center
                input[6] = Channel.of([
                    [ id:'test_fasta' ], // meta map
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true) ]
                ])
                input[7] = Channel.of([[:], []])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.orig_bam,
                    workflow.out.bam_sorted,
                    workflow.out.fastq,
                    workflow.out.tab,
                    workflow.out.orig_bam_transcript,
                    file(workflow.out.bam[0][1]).name,
                    file(workflow.out.bai[0][1]).name,
                    workflow.out.stats,
                    workflow.out.flagstat,
                    workflow.out.idxstats,
                    workflow.out.bam_transcript,
                    workflow.out.bai_transcript,
                    workflow.out.stats_transcript,
                    workflow.out.flagstat_transcript,
                    workflow.out.idxstats_transcript,
                    file(workflow.out.log_out[0][1]).name,
                    file(workflow.out.log_final[0][1]).name,
                    file(workflow.out.log_progress[0][1]).name
                ).match("single_end") }
            )
        }
    }

    test("homo_sapiens - fastq - paired_end") {
        config "./nextflow.config"

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    [
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_2.fastq.gz', checkIfExists: true)
                    ]
                ])
                input[1] = STAR_GENOMEGENERATE.out.index
                input[2] = Channel.of([
                    [ id:'test_gtf' ], // meta map
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true) ]
                ])
                input[3] = true       // star_ignore_sjdbgtf
                input[4] = 'illumina' // seq_platform
                input[5] = false      // seq_center
                input[6] = Channel.of([
                    [ id:'test_fasta' ], // meta map
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true) ]
                ])
                input[7] = Channel.of([[:], []])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.orig_bam,
                    workflow.out.bam_sorted,
                    workflow.out.fastq,
                    workflow.out.tab,
                    workflow.out.orig_bam_transcript,
                    file(workflow.out.bam[0][1]).name,
                    file(workflow.out.bai[0][1]).name,
                    workflow.out.stats,
                    workflow.out.flagstat,
                    workflow.out.idxstats,
                    workflow.out.bam_transcript,
                    workflow.out.bai_transcript,
                    workflow.out.stats_transcript,
                    workflow.out.flagstat_transcript,
                    workflow.out.idxstats_transcript,
                    file(workflow.out.log_out[0][1]).name,
                    file(workflow.out.log_final[0][1]).name,
                    file(workflow.out.log_progress[0][1]).name
                ).match("paired_end") }
            )
        }
    }

    test("homo_sapiens - fastq - paired_end - with_transcriptome") {
        config "./with_transcripts.config"

        setup {
            run("RSEM_PREPAREREFERENCE") {
                script "../../../../modules/nf-core/rsem/preparereference/main.nf"
                process {
                    """
                    input[0] = channel.of(file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkifexists: true))
                    input[1] = channel.of(file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkifexists: true))
                    """
                }
            }
        }
        when {
            workflow {
                """
                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    [
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_2.fastq.gz', checkIfExists: true)
                    ]
                ])
                input[1] = STAR_GENOMEGENERATE.out.index
                input[2] = Channel.of([
                    [ id:'test_gtf' ], // meta map
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true) ]
                ])
                input[3] = true       // star_ignore_sjdbgtf
                input[4] = 'illumina' // seq_platform
                input[5] = false      // seq_center
                input[6] = Channel.of([
                    [ id:'test_fasta' ], // meta map
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true) ]
                ])
                input[7] = RSEM_PREPAREREFERENCE.out.transcript_fasta.map{[[:], it]}
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.orig_bam,
                    workflow.out.bam_sorted,
                    workflow.out.fastq,
                    workflow.out.tab,
                    workflow.out.orig_bam_transcript,
                    file(workflow.out.bam[0][1]).name,
                    file(workflow.out.bai[0][1]).name,
                    workflow.out.stats,
                    workflow.out.flagstat,
                    workflow.out.idxstats,
                    workflow.out.bam_transcript,
                    workflow.out.bai_transcript,
                    workflow.out.stats_transcript,
                    workflow.out.flagstat_transcript,
                    workflow.out.idxstats_transcript,
                    file(workflow.out.log_out[0][1]).name,
                    file(workflow.out.log_final[0][1]).name,
                    file(workflow.out.log_progress[0][1]).name
                ).match("transcriptome") }
            )
        }
    }

    test("homo_sapiens - fastq - paired_end - transcripts - no_transcriptome") {
        config "./with_transcripts.config"

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    [
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_2.fastq.gz', checkIfExists: true)
                    ]
                ])
                input[1] = STAR_GENOMEGENERATE.out.index
                input[2] = Channel.of([
                    [ id:'test_gtf' ], // meta map
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true) ]
                ])
                input[3] = true       // star_ignore_sjdbgtf
                input[4] = 'illumina' // seq_platform
                input[5] = false      // seq_center
                input[6] = Channel.of([
                    [ id:'test_fasta' ], // meta map
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true) ]
                ])
                input[7] = Channel.of([[:], []])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.orig_bam,
                    workflow.out.bam_sorted,
                    workflow.out.fastq,
                    workflow.out.tab,
                    workflow.out.orig_bam_transcript,
                    file(workflow.out.bam[0][1]).name,
                    file(workflow.out.bai[0][1]).name,
                    workflow.out.stats,
                    workflow.out.flagstat,
                    workflow.out.idxstats,
                    workflow.out.bam_transcript,
                    workflow.out.bai_transcript,
                    workflow.out.stats_transcript,
                    workflow.out.flagstat_transcript,
                    workflow.out.idxstats_transcript,
                    file(workflow.out.log_out[0][1]).name,
                    file(workflow.out.log_final[0][1]).name,
                    file(workflow.out.log_progress[0][1]).name
                ).match("no transcriptome") }
            )
        }
    }
}
