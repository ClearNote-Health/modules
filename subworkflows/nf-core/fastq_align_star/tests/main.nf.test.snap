{
    "no transcriptome": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Aligned.out.bam:md5,984e1586c4a0ee395d1f298f8ffe2d43"
                ]
            ],
            [
                
            ],
            [
                
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.SJ.out.tab:md5,844af19ab0fc8cd9a3f75228445aca0d"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Aligned.toTranscriptome.out.bam:md5,afa64bc411f13f4de255ea73a483190b"
                ]
            ],
            "test.sorted.bam",
            "test.sorted.bam.bai",
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.sorted.bam.stats:md5,39273789abce501dc706fa94bbd0079d"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.sorted.bam.flagstat:md5,db0e25cd0b37d3030e807846c022199e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.sorted.bam.idxstats:md5,107ca94dd426cc44db316f0d402307c5"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.sorted.bam:md5,304982537be52d39f99c33e6079297ff"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.sorted.bam.bai:md5,b713ac63c6114db4482fde48a87a3b55"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.sorted.bam.stats:md5,fa2f10820ffe578ebe7d5935155b4366"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.sorted.bam.flagstat:md5,a204dee59ef8b7cd0f7d952a80119b77"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.sorted.bam.idxstats:md5,4030919b4a05393dcd7e699d72725803"
                ]
            ],
            "test.Log.out",
            "test.Log.final.out",
            "test.Log.progress.out"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-05-24T10:02:48.857080497"
    },
    "single_end": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test.Aligned.out.bam:md5,2877597d2317c27a6534db71bdcfb0dc"
                ]
            ],
            [
                
            ],
            [
                
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test.SJ.out.tab:md5,75a516ab950fb958f40b29996474949c"
                ]
            ],
            [
                
            ],
            "test.sorted.bam",
            "test.sorted.bam.bai",
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test.sorted.bam.stats:md5,2b8eb6967b68f03584cc242499658e92"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test.sorted.bam.flagstat:md5,075e7a684a7ceb5fd1dae154f823128a"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test.sorted.bam.idxstats:md5,0d5a12e3e3ffdb15f1c6c50f02f22575"
                ]
            ],
            [
                
            ],
            [
                
            ],
            [
                
            ],
            [
                
            ],
            [
                
            ],
            "test.Log.out",
            "test.Log.final.out",
            "test.Log.progress.out"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-05-24T10:00:05.935042889"
    },
    "transcriptome": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Aligned.out.bam:md5,984e1586c4a0ee395d1f298f8ffe2d43"
                ]
            ],
            [
                
            ],
            [
                
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.SJ.out.tab:md5,844af19ab0fc8cd9a3f75228445aca0d"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Aligned.toTranscriptome.out.bam:md5,afa64bc411f13f4de255ea73a483190b"
                ]
            ],
            "test.sorted.bam",
            "test.sorted.bam.bai",
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.sorted.bam.stats:md5,39273789abce501dc706fa94bbd0079d"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.sorted.bam.flagstat:md5,db0e25cd0b37d3030e807846c022199e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.sorted.bam.idxstats:md5,107ca94dd426cc44db316f0d402307c5"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.sorted.bam:md5,f4b96ef58a9e762495e8a3ed27541ddf"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.sorted.bam.bai:md5,6cf664f6eeb968efdebbc44edf5a4bcb"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.sorted.bam.stats:md5,8b4b458a77eb4ffb04a568de36ecb677"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.sorted.bam.flagstat:md5,a204dee59ef8b7cd0f7d952a80119b77"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.sorted.bam.idxstats:md5,4030919b4a05393dcd7e699d72725803"
                ]
            ],
            "test.Log.out",
            "test.Log.final.out",
            "test.Log.progress.out"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-05-24T10:01:54.522549621"
    },
    "paired_end": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Aligned.out.bam:md5,983a8900e9c1b1f86665ac40ec4df738"
                ]
            ],
            [
                
            ],
            [
                
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.SJ.out.tab:md5,844af19ab0fc8cd9a3f75228445aca0d"
                ]
            ],
            [
                
            ],
            "test.sorted.bam",
            "test.sorted.bam.bai",
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.sorted.bam.stats:md5,39273789abce501dc706fa94bbd0079d"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.sorted.bam.flagstat:md5,db0e25cd0b37d3030e807846c022199e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.sorted.bam.idxstats:md5,107ca94dd426cc44db316f0d402307c5"
                ]
            ],
            [
                
            ],
            [
                
            ],
            [
                
            ],
            [
                
            ],
            [
                
            ],
            "test.Log.out",
            "test.Log.final.out",
            "test.Log.progress.out"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-05-24T10:01:00.432031171"
    }
}